packages/presentation/lib/screen/home/home_page.dart

```diff
@@ -1,177 +1,159 @@
-    return BlocProvider(
-      create: (context) => HomeBloc(
-        blocCheckUseCase: CheckNumberUseCase.instance,
-        blocGenerateUseCase: GenerateNumberUseCase.instance,
-      ),
-      child: MaterialApp(
-        debugShowCheckedModeBanner: false,
-        title: 'Generate',
-        theme: ThemeData(
-          primarySwatch: Colors.teal,
-        ),
-        home: const HomePage(title: 'Generate numbers'),
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'Generate',
+      theme: ThemeData(
+        primarySwatch: Colors.teal,
       ),
+      home: const HomePage(),
     );
   }
 }
 
-class HomePage extends StatefulWidget {
-  const HomePage({Key? key, required this.title}) : super(key: key);
-
-  final String title;
+class HomePage extends BlocScreen {
+  const HomePage({Key? key}) : super(key: key);
 
   @override
   State<HomePage> createState() => _HomePageState();
 }
 
-class _HomePageState extends State<HomePage> {
+class _HomePageState extends BlocScreenState<HomePage, HomeBloc> {
   late TextEditingController _controller;
 
+  _HomePageState()
+      : super(HomeBloc(
+          GenerateNumberUseCase(),
+          CheckNumberUseCase(),
+        ));
+
   @override
   void initState() {
     super.initState();
     _controller = TextEditingController();
-    BlocProvider.of<HomeBloc>(context).add(GenerateNumberEvent());
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    return BlocListener<HomeBloc, HomeState>(
-      listenWhen: (_, currentState) {
-        return currentState.gameState == GameState.win ||
-            currentState.gameState == GameState.lose;
-      },
-      listener: (context, state) {
-        String titleText = '';
-        String contentText = '';
-        String buttonText = '';
-
-        if (state.gameState == GameState.lose) {
-          titleText = 'You lose!';
-          contentText = 'The attempts are over. Try again!';
-          buttonText = 'Ok :(';
-        } else if (state.gameState == GameState.win) {
-          titleText = 'You won!';
-          contentText = 'The number was : ${state.generatedNumber}';
-          buttonText = 'Generate new number';
-        }
-
-        showDialog(
+    bloc.dialogStream.listen(
+      (event) {
+        if (event is GameAlertDialog) {
+          showDialog(
             context: context,
             barrierDismissible: false,
             builder: (context) {
               return ShowDialogWidget(
-                  titleText: titleText,
-                  contentText: contentText,
-                  buttonText: buttonText,
+                  titleText: event.titleText,
+                  contentText: event.contentText,
+                  buttonText: event.buttonText,
                   widgetFunction: () {
-                    BlocProvider.of<HomeBloc>(context).add(
-                      GenerateNumberEvent(),
-                    );
+                    bloc.generate();
```

Comment:
naming - rename to smth like `startNewGame`

Reply:
I prefer this naming because it close to bloc that I used before

Reply:
yes, but this name doesn't display the logic which method contains

Reply:
corrected